### PR TITLE
[Fix][Settings] Fix the definition of the subtitles.tv and subtitles.movie service settings.

### PR DESCRIPTION
**Pull request by ace20022**

### CODE REVIEW

**Comment by Montellese** [comment]
Can you explain what the issue right now is? The change and the commit message don't fit together for me.

**Comment by MilhouseVH** [comment]
Does it solve [this](http://forum.kodi.tv/showthread.php?tid=211501&pid=1952279#pid1952279), the inability to select "None"?

**Comment by MilhouseVH** [comment]
Yes, that seems to be what this change is about - allowing "None" as a tv/movie subtitle selection (nice, thanks!)

**Comment by Montellese** [comment]
If @MilhouseVH is correct the commit message needs to be fixed.

**Comment by ace20022** [comment]
That's a side effect. Currently when no sub service is installed, the button shows "None". Clicking on it is a nop. That's what I wanted to fix actually ;)

**Comment by Montellese** [comment]
That sounds more like a bug in the dialog or not? But well TBH it doesn't make sense to not define a default value and not allow the setting to be empty. That's a contradiction in itself.

**Comment by ace20022** [comment]
> That sounds more like a bug in the dialog or not?

I'm not sure that I understand. ```CGUIWindowAddonBrowser::SelectAddonID``` is called with
```c++
showNone            false
showDetails         true
multipleSelection   false
showInstalled       true
showInstallable	    false
showMore            true

```
And since nothing is installed and showNone is false it's a nop.

Removing https://github.com/xbmc/xbmc/blob/master/xbmc/addons/GUIWindowAddonBrowser.cpp#L541 and https://github.com/xbmc/xbmc/blob/master/xbmc/addons/GUIWindowAddonBrowser.cpp#L557

would also work. A dialog without a "None" entry is shown in this case.

**Comment by ace20022** [comment]
But then you can't remove a service/ set it to None

File: system/settings/settings.xml [hunk 1]
```diff
@@ -1082,6 +1082,7 @@
           <level>1</level>
           <default>-</default>
```
Comment (Montellese) [inline system/settings/settings.xml]: While you're at it, please remove the "-" from the default value (same for `subtitles.movie`). While it's handled by the settings system it doesn't really make any sense anymore.

**Comment by Montellese** [comment]
IMO you should still change the commit message because what this does is fix the definition of the `subtitles.tv` and `subtitles.movie` settings because they are broken right now. You can't specify an empty default value but not allow empty values.

Installing subtitle services from the setting works perfectly fine already (as long as you already have a subtitle service installed).

**Comment by ace20022** [comment]
> ... works perfectly fine already (as long as you already have a subtitle service installed).

lol

I'll change it.

**Comment by mkortstiege** [comment]
This one is good to go, right?

**Comment by Montellese** [comment]
Yup. @ace20022: Thanks for adjusting the commit message.

**Pull request merged**